packages/candid/src/idl.ts

```diff
@@ -1120,6 +1120,14 @@ export class RecordClass extends ConstructType<Record<string, any>> {
     return x;
   }
 
+  get fieldsAsObject(): Record<string, Type> {
```

Comment:
Convenient, but maybe inefficient too? Also, shouldn't we be comparing the fields by their hashes, not their names? If there's a collision, something might go subtly wrong.